Package.swift

```diff
@@ -17,7 +17,13 @@ import PackageDescription
 let package = Package(
     name: "swift-openapi-generator",
     platforms: [
-        .macOS(.v10_15)
+        .macOS(.v10_15),
+
+        // The platforms below are not currently supported for running the
+        // generator itself (only the generated code), they are here
+        // to allow the generator to emit a more descriptive error,
+        // see the file PlatformChecks.swift for details.
+        .iOS(.v13), .tvOS(.v13), .watchOS(.v6),
```

Comment:
I'm really not a fan of us having to do this and would rather us consider removing `_OpenAPIGeneratorCore` from `products`.

If we do have to do it, we can make use of string version numbers for unreasonably high version numbers, e.g.

```suggestion
        .macOS(.v10_15),

        // The platforms below are not currently supported for running the generator itself.
        // We include them here to allow the us to emit a more descriptive compiler error.
        .iOS("99"), .tvOS("99"), .watchOS("99"),
```

Reply:
I'll do the high version numbers, are 9999 instead also good? I've seen that used elsewhere.

Reply:
While they work in `@available` annotations, I tried them in this setting and it caused less readable errors because there are some other macro expansions which seem to rely on the major component being 2 digits. You can try and see for yourself what happens if you use 9999.